src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/ChangeLevelCalculator.kt

```diff
@@ -11,14 +11,35 @@ class ChangeLevelCalculator(
 ) {
 
   fun calculateChangeLevel(crn: String): TierLevel<ChangeLevel> {
-    val points = getOasysNeedsPoints(crn).plus(getOgrsPoints(crn))
-    val tier = when {
-      points >= 20 -> ChangeLevel.THREE
-      points in 10..19 -> ChangeLevel.TWO
-      else -> ChangeLevel.ONE
+    if (shouldCalculateChangeLevel(crn)) {
+
+      val points = getOasysNeedsPoints(crn).plus(getOgrsPoints(crn))
+      val tier = calculateTier(points)
+
+      return TierLevel(tier, points)
+    }
+    return TierLevel(ChangeLevel.ZERO, 0)
+  }
+
+  private fun shouldCalculateChangeLevel(crn: String): Boolean {
+    val isCurrentCustodial = communityApiDataService.isCurrentCustodialSentence(crn)
```

Comment:
Not blocking, but would this be clearer using a WHEN?

```
return when {
dsffdsf -> true
dsfdsfds -> method()
else -> false
}
```

Reply:
It's definitely more Kotlin. I should probably read a book on it or something